dns-server/src/dns_server/mod.rs

```diff
@@ -70,27 +75,35 @@ const TTL_IP: u32 = 3600;
 const TTL_NS: u32 = 21600;
 const TTL_SOA: u32 = 21600;
 
-// Maximum number of IPv4 addresses in result
+/// Maximum number of IPv4 addresses in result
 const MAX_IPV4_RECORDS: usize = 24;
 
-// Maximum number of IPv6 addresses in result
+/// Maximum number of IPv6 addresses in result
 const MAX_IPV6_RECORDS: usize = 14;
 
+/// When publishing addresses, we give preference to nodes that have the same software version as
+/// the dns server itself.
+/// This constant is a number between 0 and 1 that determines how many addresses of same-version
+/// nodes will be returned compared to nodes of any other version.
+const SAME_SOFTWARE_VERSION_PEERS_RATIO: f64 = 0.8;
```

Comment:
0.95 is probably even better.

Reply:
Can we set this with a command line argument? How hard would that be?

Reply:
Btw, PerThousand can be taken from a String in both percentage and fraction. So like I said in the other command, PerThousand sounds like a better way to represent this.